### PR TITLE
fix: fetcher.submit failing with plain objects containing tagName pro…

### LINKED ISSUE #14526: fetcher.submit Fails When Plain Data Object Contains tagName Property
### Reproduction

When using `fetcher.submit` (from the useFetcher hook) to submit a plain JavaScript object as data (especially with encType: "application/json"), the submission fails with an error intended for invalid DOM elements if the data object contains a property named `tagName`.

The internal logic seems to incorrectly identify the data object as an invalid HTML element due to the presence of the `tagName` property, regardless of its value.

## Steps to Reproduce
- Use useFetcher in a component.
- Define a plain JavaScript object for submission that includes a tagName property
```ts
const formData = {
  tagName: "chatgpt", // The key 'tagName' causes the issue.
  icon: null
};
```
- Call fetcher.submit with this data and the JSON encoding
```ts
fetcher.submit(formData, {
  action: "/api/tags",
  method: "POST",
  encType: "application/json",
});
```

### System Info

```shell
System:
    OS: Linux 6.16 Pop!_OS 22.04 LTS
    CPU: (4) x64 Intel(R) Core(TM) i5-5200U CPU @ 2.20GHz
    Memory: 5.06 GB / 15.52 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 22.17.0 - /home/mzeeshan/.nvm/versions/node/v22.17.0/bin/node
    npm: 11.6.2 - /home/mzeeshan/.nvm/versions/node/v22.17.0/bin/npm
    pnpm: 10.21.0 - /home/mzeeshan/.nvm/versions/node/v22.17.0/bin/pnpm
  Browsers:
    Chrome: 140.0.7339.185
    Firefox: 143.0
    Firefox Developer Edition: 143.0
  npmPackages:
    @react-router/dev: ^7.9.2 => 7.9.5 
    @react-router/node: ^7.9.2 => 7.9.5 
    @react-router/serve: ^7.9.2 => 7.9.5 
    react-router: ^7.9.2 => 7.9.5 
    vite: ^7.1.7 => 7.2.2
```

### Used Package Manager

pnpm

### Expected Behavior

The formData object should be serialized and submitted as JSON to the route action defined at /api/tags, as encType: "application/json" is specified.

### Actual Behavior

The submission fails with the following error:
```
Cannot submit element that is not <form>, <button>, or <input type="submit|image">
```

### PR DESCRIPTION
**Pull request by joseph0926**
Fixes #14526 

## Summary

Updates `isHtmlElement()` to use `instanceof HTMLElement` instead of checking for the presence of a `tagName` string property. This prevents plain data objects with a `tagName` property from being incorrectly identified as HTML elements.

The previous implementation would throw an error when submitting objects like `{ tagName: "div", data: "test" }` via `fetcher.submit()`, as they were mistakenly treated as invalid HTML elements.

## Changes

- Updated `isHtmlElement()` in `packages/react-router/lib/dom/dom.ts` to use `instanceof HTMLElement`

- I largely followed the approach described in [rossipedia’s comment](https://github.com/remix-run/react-router/issues/14526#issuecomment-3519037126), and additionally only introduced the following tests and edge-case handling,

   - Added a guard for environments where HTMLElement is not defined (e.g., Node.js)
   - Added test cases to verify that plain objects with a tagName property are handled correctly

## Test

- Added new test file `packages/react-router/__tests__/dom/fetcher-submit-tagname-test.tsx`

### CODE REVIEW

**Comment by brophdawg11** [comment]
Thanks!  Would you mind adding a changeset via `pnpm changeset`?

**Comment by joseph0926** [comment]
> Thanks! Would you mind adding a changeset via `pnpm changeset`?

[58f17cc](https://github.com/remix-run/react-router/pull/14534/commits/58f17cc1e83b14a139febd9cf86b8565ae1e770c)

I added the changeset

File: .changeset/fluffy-walls-add.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"react-router": patch
+---
+
+Fix fetcher.submit failing with plain objects containing tagName property
```
Comment (brophdawg11) [inline .changeset/fluffy-walls-add.md]:
```suggestion
Fix `fetcher.submit` failing with plain objects containing a `tagName` property
```

**Comment by brophdawg11** [comment]
Thanks!
